clang/lib/Driver/Driver.cpp

```diff
@@ -7638,6 +7588,71 @@ Driver::getOffloadArchs(Compilation &C, const llvm::opt::DerivedArgList &Args,
       }
     }
 
+    if (Arg->getOption().matches(options::OPT_fsycl_targets_EQ)) {
```

Comment:
This should probably be protected with `Kind == Action::OFK_SYCL` as we don't want `-fsycl-targets` to populate any arch values for non-SYCL toolchains.